scipy/signal/_spline_filters.py

```diff
@@ -797,41 +797,34 @@ def symiirorder2(input, r, omega, precision=-1.0):
         The filtered signal.
     """
     xp = array_namespace(input)
-
-    # internals are numpy-only
-    input = np.asarray(input)
-    omega = np.asarray(omega)
+    input = xp_promote(input, force_floating=True, xp=xp)
+    # This function uses C internals
+    input = np.ascontiguousarray(input)
 
     if r >= 1.0:
         raise ValueError('r must be less than 1.0')
 
     if input.ndim > 2:
         raise ValueError('Input must be 1D or 2D')
 
-    if not input.flags.c_contiguous:
-        input = input.copy()
-
     squeeze_dim = False
     if input.ndim == 1:
         input = input[None, :]
         squeeze_dim = True
 
-    if np.issubdtype(input.dtype, np.integer):
-        input = input.astype(np.promote_types(input.dtype, np.float32))
-
     rsq = r * r
-    a2 = 2 * r * np.cos(omega)
+    a2 = 2 * r * math.cos(omega)
```

Comment:
Despite what the docs say, I wouldn't be surprised if historic usage allows 0D arrays or `np.float64`s. This part of the codebase is straight from the multipack era. Let's see if tests complain.